cmd/serve.go

```diff
@@ -120,3 +135,53 @@ func serve(_ context.Context) {
 		logger.Fatal("failed to run server", zap.Error(err))
 	}
 }
+
+func getAuthMiddleware(ctx context.Context, config fositex.OAuth2Configurator) (echo.MiddlewareFunc, error) {
+	issuer := config.GetAccessTokenIssuer(ctx)
+
+	audience, err := url.JoinPath(issuer, "userinfo")
+	if err != nil {
+		return nil, err
+	}
+
+	var buff bytes.Buffer
+
+	err = json.NewEncoder(&buff).Encode(config.GetSigningJWKS(ctx))
+	if err != nil {
+		return nil, err
+	}
+
+	jwks, err := keyfunc.NewJSON(json.RawMessage(buff.Bytes()))
+	if err != nil {
+		return nil, err
+	}
+
+	authConfig := echojwtx.AuthConfig{
+		Audience: audience,
+		Issuer:   issuer,
+		JWTConfig: echojwt.Config{
+			Skipper: noAuthEndpointSkipper,
+			KeyFunc: jwks.Keyfunc,
+		},
+	}
+
+	auth, err := echojwtx.NewAuth(ctx, authConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return auth.Middleware(), nil
+}
+
+func noAuthEndpointSkipper(c echo.Context) bool {
+	if echox.SkipDefaultEndpoints(c) {
+		return true
+	}
+
+	switch c.Request().URL.Path {
+	case "/token", "/jwks.json", "/.well-known/openid-configuration":
```

Comment:
Not sure I like hardcoding these paths (except for `/.well-known/openid-configuration`), but I don't really have any better ideas at the moment unless we want to make `/token` and `jwks.json` configurable.

Reply:
i've made this to where each package can expose a skipper so that the package may manage it within its own package. I think this is much cleaner and easier to maintain.